lib/tessel/deploy.js

```diff
@@ -342,854 +337,105 @@ actions.sendBundle = function(tessel, opts) {
-  var result = cp.spawnSync(python, ['-c', program]);
-  var output = result.output;
-
-  if (output == null) {
-    return '';
-  }
-
-  return output.reduce((accum, buffer) => {
-    if (buffer) {
-      accum += decoder.write(buffer);
-    }
-    return accum;
-  }, '');
-};
-
-actions.injectBinaryModules = function(globRoot, tempBundlePath, options) {
-  var binaryPathTranslations = deployLists.binaryPathTranslations['*'];
-
-  return new Promise((resolve) => {
-    options = options || {};
-
-    if (options.single) {
-      return resolve();
-    }
-
-    // For every binary module in use...
-    binaryModulesUsed.forEach(details => {
-      if (details.resolved) {
-        var translations = binaryPathTranslations.slice().concat(
-          deployLists.binaryPathTranslations[details.name] || []
-        );
-        var buildDir = details.buildPath.replace(path.dirname(details.buildPath), '');
-        var sourceBinaryPath = path.join(details.extractPath, buildDir, details.binName);
-        var tempTargetModulePath = path.join(tempBundlePath, details.modulePath);
-        var tempTargetBinaryPath = path.join(tempTargetModulePath, details.buildPath, details.binName);
-
-        sourceBinaryPath = translations.reduce((accum, translation) => {
-          return accum.replace(translation.find, translation.replace);
-        }, sourceBinaryPath);
-
-        tempTargetBinaryPath = translations.reduce((accum, translation) => {
-          return accum.replace(translation.find, translation.replace);
-        }, tempTargetBinaryPath);
-
-        fs.copySync(sourceBinaryPath, tempTargetBinaryPath);
-
-        // Also ensure that package.json was copied.
-        fs.copySync(
-          path.join(globRoot, details.modulePath, 'package.json'),
-          path.join(tempTargetModulePath, 'package.json')
-        );
-      } else {
-        if (!details.ignored) {
-          logMissingBinaryModuleWarning(details.name);
-        }
-        // In the future we may allow users to log the ignored modules here
-      }
-    });
-
-    // All binary modules have been replaced, resolve.
-    return resolve();
-  });
-};
-
-actions.tarBundle = function(opts) {
-  var cwd = process.cwd();
-  var target = opts.target || cwd;
-  var relative = path.relative(cwd, target);
-  var globRoot = relative || target;
-  var packer = tar.Pack({
-    noProprietary: true
-  });
-  var buffers = [];
+    var prom = Promise.resolve();
 
-  if (opts.full) {
-    opts.slim = false;
-  }
-
-  var includeRules = actions.glob.rules(target, '.tesselinclude');
-
-  // Convert `deployLists.includes` into includeRules
-  deployLists.includes.forEach(include => includeRules.push(`node_modules/**/${include}`));
-
-  var includeFiles = actions.glob.files(globRoot, includeRules);
-  var includeNegateRules = includeRules.reduce((rules, pattern) => {
-    if (pattern.indexOf('!') === 0) {
-      rules.push(pattern.slice(1));
+    if (opts.lang.preRun) {
```

Comment:
Will the `lang` option be defined in all cases? I ran `t2 restart` and I got the following error:
```
ERR! TypeError: Cannot read property 'preRun' of undefined
```